File: JavaScript/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    const N = numbers.length

    function dfs(number, idx) {
        if (idx == N) {
            if (number == target) {
                answer++
            }
            return
        }
        dfs(number + numbers[idx], idx + 1)
        dfs(number - numbers[idx], idx + 1)
    }

    dfs(0, 0)
    return answer;
}

File: Java/src/타겟넘버.java
//public class 타겟넘버 {
//    public static void main(String[] args) {
//        int[] numbers = {1,1,1,1,1};
//        int target = 3;
//
//        // 시작
//        int answer = 0;
//
//
//
//        int result = dfs(numbers, target,0,0);
//
//    }
//    public static int dfs(int[] numbers,int target int idx, int total) {
//        System.out.println(idx);
//        if (idx == numbers.length) {
//            if (total == target) {
//                return 1;
//            }
//            return 0;
//        }
//
//        dfs(numbers,target, idx+1, total+numbers[idx]);
//        dfs(numbers,target,idx+1, total-numbers[idx]);
//    };
//}
